Par 30/header contact us form

oiva-terho opened the pull request:
+ popup as dialog element
+ form with name, phone, note and agreement fields
+ close by button, esc key and background
+ input check and error notification
+ fetch placeholder
+ fetch result success and error messages
+ success and error svg images
+ spinner for loading
+ scroll fix while popup is opened

oiva-terho committed:
contact us popup

+ popup itself
+ close by button and background
+ form with name, phone, note and agreement fields
+ error handle
+ input check and error notification
+ fetch placeholder
+ fetch result success and error messages
+ success and error svg images
+ spinner for loading

oiva-terho committed:
fixed scroll behind the modal

oiva-terho committed:
removed redudant animation

andrey-kudinov commented on src/components/Popup/Popup.module.css:
```diff
@@ -0,0 +1,109 @@
+  transform: translate(-50%, -50%);
+  background-color: #f5f5f5;
+  color: #4c5056;
+  border: none;
```
Добавь пж скругленные уголки, на `15px`

andrey-kudinov commented on src/components/Popup/Popup.module.css:
```diff
@@ -0,0 +1,109 @@
+dialog.popup {
```
На сафари старше 2022 года открывается с глюком этот тег
Модалка появляется сразу при загрузке и её не удаётся закрыть

<img width="1028" alt="image" src="https://github.com/DevMentorship/paravina-rebuild/assets/64890665/f764b3f6-4f15-47b4-932b-c516c1887068">
https://caniuse.com/?search=dialog

Я думаю нам нужен fallback или пока что лучше просто использовать div
Я думаю с текущей поддержкой мы пока не готовы к использованию этого тега (

andrey-kudinov replied:
Без атрибута open она открыта
<img width="898" alt="image" src="https://github.com/DevMentorship/paravina-rebuild/assets/64890665/7160b725-8f4c-4acb-875f-00a6184e4759">
Я использовал для тестирования сервис browserstack.com

andrey-kudinov commented on src/components/Popup/Popup.module.css:
```diff
@@ -0,0 +1,109 @@
+  top: 0;
+  right: 0;
+  cursor: pointer;
+  padding: 1.5em;
```
Для НЕ шрифтовых свойств используй пж px
Для шрифтовых rem
В проекте 1rem = 10px

andrey-kudinov commented on src/components/Popup/Popup.module.css:
```diff
@@ -0,0 +1,109 @@
+}
+
+.popup > form > input:focus-visible, 
+.popup > form > textarea:focus-visible {
```
👍

andrey-kudinov commented on src/components/Popup/Popup.tsx:
```diff
@@ -0,0 +1,169 @@
+  function fetchSubmit(_data: IFormFields): Promise<string> {
+    //Here should be backend endpoint for submitting
+
+    return new Promise((resolve, reject) => {
+      setTimeout(() => {
+        const randomNumber = Math.random();
+        if (randomNumber < 0.1) {
+          resolve('200');
+        } else {
+          reject(new Error('400'));
+        }
+      }, 2000); // Simulating a delay of 2 seconds
+    });
```
👍

andrey-kudinov commented on src/components/Popup/Popup.tsx:
```diff
@@ -0,0 +1,169 @@
+            {submitStatus.err && <span className={cn(styles['popup-err'])}>{submitStatus.err}</span>}
+            <label>
+              <input required type="checkbox" name="agreement" checked={agreement} onChange={handleChange} />
+              <a href="/agreement">Согласие с условиями</a>
```
Здесь не очевидно что это ссылка
Ссылкой может быть только слово `Условия` и оно может быть подчёркнуто как обычная ссылка

andrey-kudinov replied:
![Uploading image.png…]()

andrey-kudinov commented on src/components/Popup/Popup.tsx:
```diff
@@ -0,0 +1,169 @@
+          <h2 className="heading2">Связаться с нами</h2>
+          <input required placeholder="Имя" name="name" type="text" value={name} onChange={handleChange} />
+          <input required placeholder="Телефон" name="phone" type="text" value={phone} onChange={handleChange} />
+          <textarea required placeholder="Примечание" name="note" value={note} onChange={handleChange} />
```
<img width="624" alt="image" src="https://github.com/DevMentorship/paravina-rebuild/assets/64890665/f48bacb3-b86e-4c97-b36b-fd31b4e8fe89">
Примечание не должно быть для нас обязательным
С нашей стороны хорошо сделать меньше препятствий для клиента, но в то же время иметь противовандальную валидацию

andrey-kudinov replied:
Например скопированный номер из тг не проходит валидацию

andrey-kudinov replied:
Здесь не отрабатывает кнопка Отправить
![Uploading image.png…]()

DmitriyAvgustinovich replied:
При открытом модальном окне на мобильных устройствах скролл все равно присутствует, ios 15.4

andrey-kudinov reviewed (changes requested):
Хорошая работа
Посмотри пж комментарии
Так же упрости пж валидацию

oiva-terho committed:
fix form validation

+ added border-radius for popup
+ changed css units of measure according to standard
+ changed agreement link display
+ phone check now accept +, -, () and whitespaces
+ note textarea is not required anymore
+ added maxlenght check on submit for textarea
+ added notification on last 50 characters and overlimit for textarea

oiva-terho commented:
@andrey-kudinov Check this solution for Safari, please. I cannot test it. Did it in blind.
I also tried https://github.com/GoogleChrome/dialog-polyfill but it failed dev server for me from the box.

oiva-terho committed:
dialog to div

+ changed modal logic from dialog element to div to fix bugs on old iOS
+ fixed scroll bug on iOS 15
+ added blur on background

oiva-terho commented:
@andrey-kudinov changed to divs, it works for me.
@DmitriyAvgustinovich fixed bug for iOS 15.

oiva-terho committed:
aria-label on close buttons

andrey-kudinov reviewed (changes requested):
Всё хорошо выглядит
Осталось немного
- убери пж черную рамку вокруг textarea
- сделай пж кнопку как остальные на сайте
- обработай пж адаптив

![Uploading image.png…]()

oiva-terho committed:
Merge branch 'devMentorship/master' of https://github.com/DevMentorship/paravina-rebuild into PAR-30/Header_Contact_Us_Form

oiva-terho committed:
styles fix

+ removed input :focus-visible border
+ removed button animation
+ close button in one line with submit button
+ small screens optimisations

oiva-terho committed:
React Hook Form added

oiva-terho committed:
removed check if modalIsOpened

oiva-terho committed:
notification styles fix

Commit:
Merge branch 'devMentorship/master' of https://github.com/DevMentorship/paravina-rebuild into PAR-30/Header_Contact_Us_Form

Commit:
Update
